complete.py

```diff
@@ -0,0 +1,771 @@
+        integrator_kwargs : dict, see default
+            kwargs to pass to OMMLIAIS integrator
+        save_indices : list(int)
+            list of indices of md_topology atoms to save to disk
+        position_extractor : function, default None
+            function to extract appropriate positons from the cache
+    """
+    from coddiwomple.particles import Particle
+    from coddiwomple.openmm.states import OpenMMParticleState, OpenMMPDFState
+    #load the endstate cache
+    traj = md.Trajectory.load(endstate_cache_filename)
+    num_frames = traj.n_frames
+    print(f"loaded {num_frames} from the cache")
+    
+    #make a handle object for ANI
+    species_str = ''.join([atom.element.symbol for atom in md_topology.subset(list(subset_indices_map.keys())).atoms])
+    print(f"species string: {species_str}")
+    ani_handler = ANI1_force_and_energy(model = torchani.models.ANI1ccx(),
+                                                 atoms=species_str,
+                                                 platform='cpu',
+                                                 temperature=integrator_kwargs['temperature'])
+
+    #make thermostates
+    pdf_state = ThermodynamicState(system = system, temperature = integrator_kwargs['temperature'], pressure= integrator_kwargs['pressure'])
+    pdf_state_subset = ThermodynamicState(system = system_subset, temperature = integrator_kwargs['temperature'], pressure = None)
+    
+    #make a reporter
+    reporter = OpenMMReporter(directory_name, trajectory_prefix, md_topology, subset_indices = save_indices)
+    
+    
+    #make an integrator
+    integrator = Integrator(**integrator_kwargs)
+    
+    #make a propagator
+    propagator = Propagator(openmm_pdf_state = pdf_state,
+                 openmm_pdf_state_subset = pdf_state_subset,
+                 subset_indices_map = subset_indices_map,
+                 integrator = integrator,
+                 ani_handler = ani_handler,
+                 context_cache=None,
+                 reassign_velocities=True,
+                 n_restart_attempts=0,
+                 reporter = reporter)
+
+    frames = np.random.choice(range(num_frames), number_of_applications)
+    particle = Particle(0)
+    
+    for i in tqdm.trange(number_of_applications):
+        if position_extractor is not None:
+            positions = position_extractor(traj.xyz[frames[i]] * unit.nanometers)
+        else:
+            positions = traj.xyz[frames[i]] * unit.nanometers
+        particle_state = OpenMMParticleState(positions = positions, box_vectors = traj.unitcell_vectors[frames[i]]*unit.nanometers)
+        particle.update_state(particle_state)
+        #try:
+        _, _return_dict = propagator.apply(particle_state, n_steps = steps_per_application, reset_integrator=True, apply_pdf_to_context=True)
+        #except Exception as e:
+        #    print(e)
+
+
+    return propagator.state_works
+
+
+# In[ ]:
+
+
+#Example calculation: let's do benzene --> methylbenzene
```

Comment:
can remove commented out code block if unused or move to a `demos` folder or similar if used